src/server.cpp

```diff
@@ -917,9 +919,19 @@ void Server::AsyncRunStep(bool initial_step)
 						&far_players, disable_single_change_sending ? 5 : 30);
 				break;
 			case MEET_BLOCK_NODE_METADATA_CHANGED:
-				infostream << "Server: MEET_BLOCK_NODE_METADATA_CHANGED" << std::endl;
+				{
+					infostream << "Server: MEET_BLOCK_NODE_METADATA_CHANGED" << std::endl;
 						prof.add("MEET_BLOCK_NODE_METADATA_CHANGED", 1);
-						setBlockNotSent(event->p);
+					//Dont send the change yet. Collect them to eliminate dupes.
+					node_meta_updates.remove(event->p);
+					node_meta_updates.push_back(event->p);
+					MapBlock *block = m_env->getMap().getBlockNoCreateNoEx(
```

Comment:
you mean like this?

```C++
MapBlock *block;
if (block = m_env->getMap().getBlockNoCreateNoEx(
		getNodeBlockPos(event->p))) {
[...]
```

That wouldn't spare lines and IMO it would make the code less readable.